File: Scripty_kurz/5_Claude_Agent_SDK/python/1_single_agent/1b_local_model.py
#!/usr/bin/env python3
"""
Local Model Demo — query() and ClaudeSDKClient with Ollama (gpt-oss)
=====================================================================
Demonstrates how to use a local Ollama model with both SDK interfaces.

Setup:
    ollama pull gpt-oss
    OLLAMA_CONTEXT_LENGTH=32768 ollama serve
"""
import os
import anyio
from claude_agent_sdk import (
    query,
    ClaudeSDKClient,
    ClaudeAgentOptions,
    AssistantMessage,
    TextBlock,
    ResultMessage,
)

# Point the SDK to your local Ollama instance
os.environ["ANTHROPIC_BASE_URL"] = "http://localhost:11434"
os.environ["ANTHROPIC_AUTH_TOKEN"] = "ollama"
os.environ["ANTHROPIC_API_KEY"] = ""

LOCAL_MODEL = "gpt-oss:20b"


# ---------------------------------------------------------------------------
# Example 1 — query(): simple, stateless, one-shot
# ---------------------------------------------------------------------------
async def example_query():
    print("=== Example 1: query() — Simple One-Shot ===\n")

    options = ClaudeAgentOptions(
        model=LOCAL_MODEL,
        system_prompt="You are a helpful assistant.",
        max_turns=1,
    )

    async for message in query(
        prompt="Tell me about Python programming.",
        options=options,
    ):
        print("-----")
        print(message)
        print("-----\n")

        if isinstance(message, AssistantMessage):
            for block in message.content:
                if isinstance(block, TextBlock):
                    print(f"gpt-oss: {block.text}")
        elif isinstance(message, ResultMessage):
            print(f"Turns: {message.num_turns}")

    print()


# ---------------------------------------------------------------------------
# Example 2 — ClaudeSDKClient: stateful, multi-turn, full-featured
# ---------------------------------------------------------------------------
async def example_client():
    print("=== Example 2: ClaudeSDKClient — Stateful Session ===\n")

    options = ClaudeAgentOptions(
        model=LOCAL_MODEL,
        system_prompt="You are a helpful assistant.",
        max_turns=3,
    )

    async with ClaudeSDKClient(options=options) as client:
        print("Asking: What is 2 + 2?\n")
        await client.query("What is 2 + 2?")

        async for message in client.receive_response():
            print("-----")
            print(message)
            print("-----\n")

            if isinstance(message, AssistantMessage):
                for block in message.content:
                    if isinstance(block, TextBlock):
                        print(f"gpt-oss: {block.text}")
            elif isinstance(message, ResultMessage):
                if message.total_cost_usd and message.total_cost_usd > 0:
                    print(f"Cost: ${message.total_cost_usd:.4f}")
                print(f"Duration: {message.duration_ms}ms")

        # Follow-up — session remembers the previous exchange
        print("\nFollow-up: Double that result.\n")
        await client.query("Double that result.")

        async for message in client.receive_response():
            if isinstance(message, AssistantMessage):
                for block in message.content:
                    if isinstance(block, TextBlock):
                        print(f"gpt-oss: {block.text}")
            elif isinstance(message, ResultMessage):
                print(f"Duration: {message.duration_ms}ms")

    print()


# ---------------------------------------------------------------------------
# Main
# ---------------------------------------------------------------------------
async def main():
    await example_query()
    await example_client()


if __name__ == "__main__":
    anyio.run(main)

File: podklady/5_Claude_Agent_SDK/typescript/src/1_single_agent/1b_local_model.ts
#!/usr/bin/env node
/**
 * Local Model Demo — query() with Ollama (gpt-oss)
 *
 * Demonstrates how to use a local Ollama model with both
 * one-shot and multi-turn patterns.
 *
 * Setup:
 *   ollama pull gpt-oss
 *   OLLAMA_CONTEXT_LENGTH=32768 ollama serve
 */

import {
  query,
  SDKAssistantMessage,
  SDKResultMessage,
  SDKUserMessage,
} from '@anthropic-ai/claude-agent-sdk';

// Point the SDK to your local Ollama instance
process.env.ANTHROPIC_BASE_URL = 'http://localhost:11434';
process.env.ANTHROPIC_AUTH_TOKEN = 'ollama';
process.env.ANTHROPIC_API_KEY = '';

const LOCAL_MODEL = 'gpt-oss:20b';

// ---------------------------------------------------------------------------
// Example 1 — query(): simple, stateless, one-shot
// ---------------------------------------------------------------------------
async function exampleQuery() {
  console.log('=== Example 1: query() — Simple One-Shot ===\n');

  for await (const message of query({
    prompt: 'Tell me about Python programming.',
    options: {
      model: LOCAL_MODEL,
      systemPrompt: 'You are a helpful assistant.',
      maxTurns: 1,
    },
  })) {
    if (message.type === 'assistant') {
      const assistantMsg = message as SDKAssistantMessage;
      for (const block of assistantMsg.message.content) {
        if (block.type === 'text') {
          console.log(`gpt-oss: ${block.text}`);
        }
      }
    } else if (message.type === 'result') {
      const resultMsg = message as SDKResultMessage;
      console.log(`Turns: ${resultMsg.num_turns}`);
    }
  }

  console.log('\n');
}

// ---------------------------------------------------------------------------
// Example 2 — Multi-turn conversation with memory
// ---------------------------------------------------------------------------
async function exampleMultiTurn() {
  console.log('=== Example 2: Multi-Turn Conversation ===\n');

  async function* conversation(): AsyncGenerator<SDKUserMessage> {
    yield {
      type: 'user',
      message: {
        role: 'user',
        content: [{ type: 'text', text: 'What is 2 + 2?' }],
      },
    };

    await new Promise((resolve) => setTimeout(resolve, 100));

    yield {
      type: 'user',
      message: {
        role: 'user',
        content: [{ type: 'text', text: 'Double that result.' }],
      },
    };
  }

  for await (const message of query({
    prompt: conversation(),
    options: {
      model: LOCAL_MODEL,
      systemPrompt: 'You are a helpful assistant.',
      maxTurns: 3,
    },
  })) {
    if (message.type === 'assistant') {
      const assistantMsg = message as SDKAssistantMessage;
      for (const block of assistantMsg.message.content) {
        if (block.type === 'text') {
          console.log(`gpt-oss: ${block.text}`);
        }
      }
    } else if (message.type === 'result') {
      const resultMsg = message as SDKResultMessage;
      if (resultMsg.total_cost_usd && resultMsg.total_cost_usd > 0) {
        console.log(`Cost: $${resultMsg.total_cost_usd.toFixed(4)}`);
      }
      console.log(`Duration: ${resultMsg.duration_ms}ms`);
    }
  }

  console.log('\n');
}

// ---------------------------------------------------------------------------
// Main
// ---------------------------------------------------------------------------
async function main() {
  await exampleQuery();
  await exampleMultiTurn();
}

main().catch(console.error);
